### PR TITLE
adding terser

### PR DESCRIPTION
**Pull request by prateekbh**
**What kind of change does this PR introduce?**

Adds terser as the minifier tool instead of Uglify

**Did you add tests for your changes?**
No

**Summary**
Terser adds support for minifying modern code like ES6/7/8, whereas uglify is no longer maintained.

**Does this PR introduce a breaking change?**
No

**Note**: I saw bigger bundles with this vs the Uglify, @lukeed can you please check the config?

### CODE REVIEW

**Comment by ForsakenHarmony** [comment]
https://github.com/developit/microbundle/pull/137#issuecomment-427982896

see this on sizes

**Comment by prateekbh** [comment]
I'll post a comparison of our base examples

**Comment by prateekbh** [comment]
Here's a result:

##Default Template
====================
**Uglify:**
```
bundle.a890a.css ⏤  445 B (+445 B)
bundle.5f360.js ⏤  7.61 kB (+7.61 kB)
polyfills.8a2ee.js ⏤  1.99 kB (+1.99 kB)
2.chunk.45507.css ⏤  74 B (+74 B)
route-home.chunk.5702a.js ⏤  296 B (+296 B)
3.chunk.4c621.css ⏤  77 B (+77 B)
route-profile.chunk.7a871.js ⏤  819 B (+819 B)
index.html ⏤  967 B (+967 B)
```


**Terser:**
```
bundle.a890a.css ⏤  445 B (+445 B)
bundle.5f360.js ⏤  7.57 kB (+7.57 kB)
polyfills.8a2ee.js ⏤  1.99 kB (+1.99 kB)
chunk.45507.css ⏤  74 B (+74 B)
chunk.5702a.js ⏤  296 B (+296 B)
chunk.4c621.css ⏤  77 B (+77 B)
chunk.7a871.js ⏤  819 B (+819 B)                     
index.html ⏤  967 B (+967 B)
```


##Material Template
**Uglify:**
```
bundle.45b77.css ⏤  4.98 kB (+4.98 kB)
bundle.9a2a6.js ⏤  24.9 kB (+24.9 kB)
polyfills.8a2ee.js ⏤  1.99 kB (+1.99 kB)
2.chunk.9e33f.css ⏤  2.36 kB (+2.36 kB)
route-home.chunk.f82a1.js ⏤  1.46 kB (+1.46 kB)
3.chunk.7ce15.css ⏤  1.69 kB (+1.69 kB)
route-profile.chunk.720cc.js ⏤  863 B (+863 B)
4.chunk.32b0f.css ⏤  1.4 kB (+1.4 kB)
route-404.chunk.9ef99.js ⏤  1.22 kB (+1.22 kB)
index.html ⏤  3.56 kB (+3.56 kB)
```

**Terser:**
```
bundle.45b77.css ⏤  4.98 kB (+4.98 kB)
bundle.9a2a6.js ⏤  26.4 kB (+26.4 kB)
polyfills.8a2ee.js ⏤  1.99 kB (+1.99 kB)
2.chunk.9e33f.css ⏤  2.36 kB (+2.36 kB)
route-home.chunk.f82a1.js ⏤  1.46 kB (+1.46 kB)
3.chunk.7ce15.css ⏤  1.69 kB (+1.69 kB)
route-profile.chunk.720cc.js ⏤  863 B (+863 B)
4.chunk.32b0f.css ⏤  1.4 kB (+1.4 kB)
route-404.chunk.9ef99.js ⏤  1.22 kB (+1.22 kB)
index.html ⏤  3.56 kB (+3.56 kB)
```

**Comment by lukeed** [comment]
So.. +4 bytes on the default template and +1.5 kB on the material template?

You removed a bunch of compression settings. I'd say this is expected.

Also, why remove?

**Comment by prateekbh** [comment]
The ones i removed are the ones that I saw as default in Terser. Do you see any which is not true?

**Comment by lukeed** [comment]
***Default: Terser***

```
              0.chunk.847b9.css ⏤  74 B (+74 B)
      route-home.chunk.5647d.js ⏤  296 B (+296 B)
              1.chunk.60d9f.css ⏤  77 B (+77 B)
   route-profile.chunk.08aae.js ⏤  819 B (+819 B)
             polyfills.de59c.js ⏤  1.85 kB (+1.85 kB)
               bundle.d9b8f.css ⏤  445 B (+445 B)
                bundle.d4425.js ⏤  7.41 kB (+7.41 kB)
                     index.html ⏤  969 B (+969 B)
```

***Default: Next***

```
              0.chunk.847b9.css ⏤  74 B (+74 B)
      route-home.chunk.5647d.js ⏤  296 B (+296 B)
              1.chunk.60d9f.css ⏤  77 B (+77 B)
   route-profile.chunk.08aae.js ⏤  819 B (+819 B)
             polyfills.de59c.js ⏤  1.85 kB (+1.85 kB)
               bundle.d9b8f.css ⏤  445 B (+445 B)
                bundle.d4425.js ⏤  7.45 kB (+7.45 kB)
                     index.html ⏤  969 B (+969 B)
```

***Default: Terser w/o Config Changes***

```
              0.chunk.847b9.css ⏤  74 B (+74 B)
      route-home.chunk.5647d.js ⏤  296 B (+296 B)
              1.chunk.60d9f.css ⏤  77 B (+77 B)
   route-profile.chunk.08aae.js ⏤  819 B (+819 B)
             polyfills.de59c.js ⏤  1.85 kB (+1.85 kB)
               bundle.d9b8f.css ⏤  445 B (+445 B)
                bundle.d4425.js ⏤  7.41 kB (+7.41 kB)
                     index.html ⏤  969 B (+969 B)
```

---

Not sure which option(s), but I'd say no need to change them. As you can see, the ~40 bytes are recovered.

File: packages/cli/lib/lib/webpack/webpack-client-config.js [hunk 1]
```diff
@@ -117,34 +117,14 @@ function isProd(config) {
...
 		optimization: {
 			minimizer: [
-				new UglifyJsPlugin({
+				new TerserPlugin({
 					cache: true,
 					parallel: true,
-					uglifyOptions: {
-						sourceMap: true,
-						output: { comments:false },
-						mangle: true,
```
[the remark this answers is not in the document]
Comment (prateekbh) [reply packages/cli/lib/lib/webpack/webpack-client-config.js]: done

**Comment by prateekbh** [comment]
This does bring the default template back to normal, not the material one though.

**Comment by prateekbh** [comment]
ahhh fixed. 
Thanks @kristoferbaxter 😆

**Pull request merged**